src/js/common-event-bindings.js

```diff
@@ -0,0 +1,95 @@
+/* global fluid */
+(function () {
+    "use strict";
+
+    /**
+     * GPII Binder Markup Events Grade
+     * This grade allows binding typical HTML events such as
+     * mouse clicks and keypress events to selectors each time
+     * the markup is rendered for a component. (Meaning they will
+     * also apply if a component is rerendered after a model refresh
+     * or similar situation)
+     *
+     * It adds a new area to a grades options called `markupEventBindings`
+     * which allows binding `selectors` to jQuery events. Theoretically
+     * other event constructs could be supported in the future, but only
+     * jQuery events are implemented at the time of writing.
+     *
+     * Binding is initiated when the components `onMarkupRendered` event is
+     * fired. If you are using a grade derived from `gpii.handlebars.templateAware`
+     * this event will be automatically fired when the handlebars markup is
+     * rendered.
+     *
+     * Example usage of adding a click handler to a selector productListLinks.
+     * ```
+     * markupEventBindings: {
+     *     productListLinks: {
+     *         // type: jQuery <- Defaults to jQuery but could be configured ITF
+     *         method: "click",
+     *         args: ["{that}.selectProduct"]
+     *     }
+     * }
+     * ```
+     */
+    fluid.defaults("gpii.binder.bindMarkupEvents", {
+        mergePolicy: {
+            decorators: "noexpand"
+        },
+        events: {
+            onDomBind: null,
+            onDomUnbind: null,
+            onMarkupRendered: null
+        },
+        listeners: {
+            onMarkupRendered: "{that}.events.onDomBind.fire({that}, {that}.container)",
+            onDestroy: "{that}.events.onDomUnbind.fire({that}, {that}.container)",
+            onDomBind: "fluid.decoratorViewComponent.processDecorators({that}, {that}.options.markupEventBindings)"
+        }
+    });
+
+    fluid.registerNamespace("fluid.decoratorViewComponent");
+
+    //
+    // The methods below might be generic enough to go straight to infusion
+    //
+
+    fluid.expandCompoundArg = function (that, arg, name) {
```

Comment:
I see your note explaining why these are in the fluid namespace.  However, they all really need JSDocs IMO.

Reply:
Add JSDocs